test/mongoose_graphql_SUITE.erl

```diff
@@ -461,9 +471,77 @@ check_child_object_field_domain_permissions(Config) ->
+    ProtectedNotEnaugh = <<"{ obj { protectedDomainName(domain: \"domain.com\") } }">>,
+    ?assertPermissionsSuccess(Config, Domain, OkDomain),
+    ?assertPermissionsSuccess(Config, Domain, OkDomain1),
+    ?assertPermissionsSuccess(Config, Domain, OkDomain2),
+    % Field is protected in interface and object, so cannot be accessed with the wrong domain.
+    ?assertDomainPermissionsFailed(Config, Domain, [<<"domain">>], WrongDomain),
+    ?assertDomainPermissionsFailed(Config, Domain, [<<"domain">>], WrongDomain1),
+    % Field is protected only in an interface, so can be accessed from implementing objects
+    % with the wrong domain.
+    ?assertPermissionsSuccess(Config, Domain, ProtectedNotEnaugh).
+
+check_field_input_arg_domain_permissions(Config) ->
+    Domain = <<"my-domain.com">>,
+    DomainInput = #{<<"domain">> => Domain, <<"notDomain">> => <<"random text here">>},
+    Config2 = [{op, <<"Q1">>}, {args, #{<<"domain">> => Domain,
+                                        <<"domainInput">> => DomainInput}} | Config],
+    Doc = <<"query Q1($domain: String, $domainInput: DomainInput!) "
+            "{ domainInputProtectedField(argA: $domain, argB: $domainInput)"
+            "  domainProtectedField(argA: $domain, argB: \"domain.com\") }">>,
+
+    FDoc = <<"{ domainInputProtectedField(argA: \"do.com\", argB: { domain: \"do.com\" }) }">>,
+    ?assertPermissionsSuccess(Config2, Domain, Doc),
+    ?assertDomainPermissionsFailed(Config, Domain, [<<"argA">>, <<"argB.domain">>], FDoc).
+
+
+check_field_list_arg_domain_permissions(Config) ->
+    Domain = <<"my-domain.com">>,
+    Domains = [#{<<"domain">> => Domain, <<"notDomain">> => <<"random text here">>},
+               #{<<"domain">> => <<"muc.", Domain/binary>>,
+                 <<"domains">> => #{<<"domain">> => Domain}}],
+    Config2 = [{op, <<"Q1">>}, {args, #{<<"domains">> => Domains}} | Config],
+    Doc = <<"query Q1($domains: [DomainInput!]) "
+            "{ domainListInputProtectedField(domains: $domains) }">>,
+
+    FDoc = <<"{ domainListInputProtectedField(domains: [{ domain: \"do.com\" }]) }">>,
+    ?assertPermissionsSuccess(Config2, Domain, Doc),
+    ?assertDomainPermissionsFailed(Config, Domain, [<<"domains.domain">>], FDoc).
+
+check_field_jid_arg_domain_permissions(Config) ->
+    Domain = <<"my-domain.com">>,
+    Config2 = [{op, <<"Q1">>},
+               {args, #{<<"jid">> => <<"bob@", Domain/binary>>}} | Config],
+    Doc = <<"query Q1($jid: JID) { domainJIDProtectedField(argA: $jid, argB: \"bob@bob\") }">>,
+    FDoc = <<"{ domainJIDProtectedField(argA: \"bob@do.com\", argB: \"bob@do.com\") }">>,
+    ?assertPermissionsSuccess(Config2, Domain, Doc),
+    ?assertDomainPermissionsFailed(Config, Domain, [<<"argA">>], FDoc).
+
+check_field_subdomain_permissions(Config) ->
+    Domain = <<"my-domain.com">>,
+    Subdomain = <<"test.", Domain/binary>>,
+    FSubdomain = <<"test.1", Domain/binary>>,
```

Comment:
Note: When the code uses `mongoose_domain_api` instead of matching strings, one would need to register the subdomain for this test to pass.